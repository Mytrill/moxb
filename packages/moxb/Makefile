all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
