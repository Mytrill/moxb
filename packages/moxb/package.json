{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
