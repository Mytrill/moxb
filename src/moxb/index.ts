export * from './impl';
export * from './meteor';
export * from './types';
