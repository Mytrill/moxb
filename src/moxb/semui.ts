export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
