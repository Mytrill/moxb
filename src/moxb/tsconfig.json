{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
