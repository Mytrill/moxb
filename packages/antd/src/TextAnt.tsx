import { ColProps } from 'antd/lib/grid';
import { observer } from 'mobx-react';
import { CSSProperties } from 'react';
import * as React from 'react';
import { labelWithHelp, parseProps } from './BindAnt';
import { Input, Form } from 'antd';
import { InputProps } from 'antd/lib/input';
import { Text } from '@moxb/moxb';
import { FormItemProps } from 'antd/lib/form/FormItem';

export interface BindStringAntProps extends React.HTMLProps<HTMLFormElement> {
    operation: Text;
    useDoubleClickToEdit?: boolean;
    help?: string;
    formStyle?: CSSProperties;
    onPressEnter?(): void;
    wrapperCol?: ColProps;
    labelCol?: ColProps;
}

export interface BindSearchStringAntProps extends BindStringAntProps {
    enterButton?: string;
    onSearch?(): void;
}

@observer
export class TextAnt extends React.Component<InputProps & BindStringAntProps> {
    // tslint:disable-next-line:cyclomatic-complexity
    render() {
        const { operation, id, inputType, value, size, prefix, invisible, onPressEnter, ...props } = parseProps(
            this.props,
            this.props.operation
        );
        if (invisible) {
            return null;
        }
        if ((inputType || operation.inputType) === 'textarea') {
            return (
                <Input.TextArea
                    id={id}
                    placeholder={operation.placeholder}
                    onFocus={operation.onEnterField}
                    onBlur={operation.onExitField}
                    value={operation.value || value || ''}
                    prefix={prefix}
                    onChange={(e: any) => operation.setValue(e.target.value)}
                    onPressEnter={onPressEnter}
                    rows={this.props.rows}
                    {...props as any}
                />
            );
        } else {
            return (
                <Input
                    id={id}
                    placeholder={operation.placeholder}
                    onFocus={operation.onEnterField}
                    onBlur={operation.onExitField}
                    value={operation.value || value || ''}
                    prefix={prefix}
                    onChange={(e: any) => operation.setValue(e.target.value)}
                    onPressEnter={onPressEnter}
                    size={size}
                    {...props as any}
                />
            );
        }
    }
}

/**
 * Use `onSearch` for the search function! Then it will work with the Enter as well as with the search button!
 */
@observer
export class TextSearchAnt extends React.Component<InputProps & BindSearchStringAntProps> {
    render() {
        const Search = Input.Search;
        const { operation, id, value, invisible, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Search
                id={id}
                placeholder={operation.placeholder}
                onFocus={operation.onEnterField}
                onBlur={operation.onExitField}
                value={operation.value || value || ''}
                onChange={(e: any) => operation.setValue(e.target.value)}
                {...props as any}
            />
        );
    }
}

@observer
export class TextFormAnt extends React.Component<FormItemProps & BindStringAntProps> {
    render() {
        const {
            operation,
            label,
            invisible,
            prefix,
            formStyle,
            labelCol,
            wrapperCol,
            onPressEnter,
            ...props
        } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Form.Item
                label={labelWithHelp(label != null ? label : operation.label, operation.help)}
                style={formStyle || undefined}
                labelCol={labelCol}
                wrapperCol={wrapperCol}
            >
                <TextAnt operation={operation} prefix={prefix} onPressEnter={onPressEnter} {...props as any} />
            </Form.Item>
        );
    }
}
