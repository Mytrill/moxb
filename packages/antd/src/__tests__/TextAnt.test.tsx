import { TextImpl } from '@moxb/moxb';
import * as React from 'react';
import { TextAnt } from '../TextAnt';
import * as Renderer from 'react-test-renderer';

describe('TextAnt', function() {
    it('should render text by default', function() {
        const operation = new TextImpl({
            id: 'TheId',
            label: 'oh',
            disabled: () => false,
        });

        // operation.setError('oops');
        // expect(ReactTestRenderer.create(<ActionFormButtonAnt operation={operation} />)).toMatchSnapshot();
        expect(Renderer.create(<TextAnt operation={operation} />)).toMatchSnapshot();
    });
    it('should render textarea by default', function() {
        const operation = new TextImpl({
            id: 'TheId',
            label: 'oh',
            control: 'textarea',
            disabled: () => false,
        });

        // operation.setError('oops');
        // expect(ReactTestRenderer.create(<ActionFormButtonAnt operation={operation} />)).toMatchSnapshot();
        expect(Renderer.create(<TextAnt operation={operation} />)).toMatchSnapshot();
    });
});
