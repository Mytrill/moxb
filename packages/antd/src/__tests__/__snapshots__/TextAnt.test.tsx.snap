// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAnt should render a text input by default 1`] = `
<Input
  disabled={false}
  id="the_id"
  label="oh"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  prefixCls="ant-input"
  readOnly={false}
  type="text"
  value=""
/>
`;

exports[`TextAnt should render a textarea with the control option set 1`] = `
<TextArea
  disabled={false}
  id="the_id"
  label="oh"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  prefixCls="ant-input"
  readOnly={false}
  value=""
/>
`;

exports[`TextFormAnt should render form item with a text input by default 1`] = `
<FormItemAnt
  operation={
    bound TextImpl {
      "_errors": Array [],
      "_isSaving": false,
      "_value": undefined,
      "constructor": [Function],
      "doClearErrors": [Function],
      "doGetValue": [Function],
      "doOnValidate": [Function],
      "doPlaceholder": [Function],
      "doSetValue": [Function],
      "domId": "the_id",
      "getDisabled": [Function],
      "getErrors": [Function],
      "getHelp": [Function],
      "getInitialValue": [Function],
      "getInvisible": [Function],
      "getLabel": [Function],
      "getReadonly": [Function],
      "id": "TheId",
      "impl": Object {
        "disabled": [Function],
        "id": "TheId",
        "label": "oh",
      },
      "setError": [Function],
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "TextImpl@86.keys",
          "observers": Set {},
        },
        "name": "TextImpl@86",
        "target": [Circular],
        "values": Map {
          "_errors" => Array [],
          "label" => "oh",
          "disabled" => false,
          "enabled" => true,
          "invisible" => false,
          "readOnly" => false,
          "help" => undefined,
          "errors" => Array [],
          "error" => undefined,
          "hasErrors" => false,
          "_value" => undefined,
          "_isSaving" => false,
          "inputType" => undefined,
          "value" => undefined,
          "placeholder" => undefined,
          "required" => undefined,
          "isInitialValue" => undefined,
          "control" => "input",
        },
      },
    }
  }
>
  <TextAnt
    disabled={false}
    id="the_id"
    operation={
      bound TextImpl {
        "_errors": Array [],
        "_isSaving": false,
        "_value": undefined,
        "constructor": [Function],
        "doClearErrors": [Function],
        "doGetValue": [Function],
        "doOnValidate": [Function],
        "doPlaceholder": [Function],
        "doSetValue": [Function],
        "domId": "the_id",
        "getDisabled": [Function],
        "getErrors": [Function],
        "getHelp": [Function],
        "getInitialValue": [Function],
        "getInvisible": [Function],
        "getLabel": [Function],
        "getReadonly": [Function],
        "id": "TheId",
        "impl": Object {
          "disabled": [Function],
          "id": "TheId",
          "label": "oh",
        },
        "setError": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "TextImpl@86.keys",
            "observers": Set {},
          },
          "name": "TextImpl@86",
          "target": [Circular],
          "values": Map {
            "_errors" => Array [],
            "label" => "oh",
            "disabled" => false,
            "enabled" => true,
            "invisible" => false,
            "readOnly" => false,
            "help" => undefined,
            "errors" => Array [],
            "error" => undefined,
            "hasErrors" => false,
            "_value" => undefined,
            "_isSaving" => false,
            "inputType" => undefined,
            "value" => undefined,
            "placeholder" => undefined,
            "required" => undefined,
            "isInitialValue" => undefined,
            "control" => "input",
          },
        },
      }
    }
    readOnly={false}
  />
</FormItemAnt>
`;

exports[`TextSearchAnt should render search text input by default 1`] = `
<Search
  disabled={false}
  enterButton={false}
  id="the_id"
  inputPrefixCls="ant-input"
  label="oh"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  prefixCls="ant-input-search"
  readOnly={false}
  value=""
/>
`;
